tuf/api/metadata.py

```diff
@@ -463,22 +463,64 @@ def remove_key(self, role: str, keyid: str) -> None:
             del self.keys[keyid]
 
 
+class MetadataInfo:
+    """A container with information about a particular metadata file.
+    Instances of MetadataInfo are used as values in a dictionary called
+    "meta" in Timestamp and Snapshot.
+    Attributes:
+        version: An integer indicating the version of the metadata file.
+        length: An optional integer indicating the length of the metadata file.
+        hashes: An optional dictionary mapping hash algorithms to the
+            hashes resulting from applying them over the metadata file
+            contents.::
+
+                'hashes': {
+                    '<HASH ALGO 1>': '<METADATA FILE HASH 1>',
+                    '<HASH ALGO 2>': '<METADATA FILE HASH 2>',
+                    ...
+                }
+
+    """
+
+    def __init__(
+        self,
+        version: int,
+        length: Optional[int] = None,
+        hashes: Optional[Dict] = None,
+    ) -> None:
+        self.version = version
+        self.length = length
+        self.hashes = hashes
+
+    def __eq__(self, other: "MetadataInfo") -> bool:
+        """Compare objects by their values instead of by their identities."""
+        return (
+            self.version == other.version
+            and self.length == other.length
+            and self.hashes == other.hashes
+        )
+
+    def to_dict(self) -> Dict:
+        """Returns the dictionary representation of self. """
+        json_dict = {"version": self.version}
```

Comment:
After abstracting the (de)serialization, other formats than `json` are possible and although the default is `json` it shouldn't be hardcoded.